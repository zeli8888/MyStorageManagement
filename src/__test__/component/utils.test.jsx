import { describe, expect, test, vi } from 'vitest';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import {
    DeletionConfirmationComponent,
    SearchComponent,
    RememberMeCheckbox,
    getVisibleRows,
    handleClick,
    EnhancedTableHead,
    EnhancedTableToolbar,
    NumberInput,
    PieChartCard,
    pieChartColors
} from '../../component/utils';
import '@testing-library/jest-dom/vitest';
import userEvent from '@testing-library/user-event';
// Mock data for table tests
const mockHeadCells = [{ id: 'name', label: 'Name', sortingEnabled: true }];
const mockRows = [{ id: 1, name: 'Apple' }, { id: 2, name: 'Banana' }];

vi.mock('react-number-format', () => ({
    NumericFormat: vi.fn(({ customInput, ...props }) => {
        const InputComponent = customInput || 'input'
        return <InputComponent {...props} />
    })
}))

describe('NumberInput component', () => {
    test('renders with default value', () => {
        render(<NumberInput defaultValue={123.45} />)
        const input = screen.getByRole('textbox')
        expect(input).toHaveValue('123.45')
    })

    // This test fails because NumericFormat doesn't trigger some how.
    // test('updates value', async () => {
    //     const user = userEvent.setup();
    //     render(<NumberInput defaultValue={1} prefix="â‚¬" />);
    //     const input = screen.getByRole('textbox');

    //     screen.debug();

    //     await user.click(input);
    //     await user.keyboard('{Control>}a{/Control}');
    //     await user.keyboard('{Backspace}');

    //     screen.debug();

    //     await user.type(input, '123.45');

    //     screen.debug();

    //     await waitFor(() => {
    //         expect(input).toHaveValue('â‚¬123.45');
    //     }, { timeout: 3000 });
    // });

    test('has empty value', () => {
        render(<NumberInput />)
        const input = screen.getByRole('textbox')
        expect(input).toHaveValue('')
    })
})

describe('DeletionConfirmationComponent', () => {
    test('renders dialog when open is true', () => {
        render(<DeletionConfirmationComponent open={true} onClose={vi.fn()}
            onConfirm={vi.fn()} warningMessage="Test message" />);
        expect(screen.getByText('Confirmation')).toBeInTheDocument();
        expect(screen.getByText('Test message')).toBeInTheDocument();
    });

    test('calls onConfirm when Yes button is clicked', () => {
        const mockConfirm = vi.fn();
        render(<DeletionConfirmationComponent open={true} onClose={vi.fn()}
            onConfirm={mockConfirm} warningMessage="Test" />);
        fireEvent.click(screen.getByText('Yes'));
        expect(mockConfirm).toHaveBeenCalled();
    });


    test('matches dialog open snapshot', () => {
        const { asFragment } = render(
            <DeletionConfirmationComponent
                open={true}
                onClose={vi.fn()}
                onConfirm={vi.fn()}
                warningMessage="Snapshot test"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('SearchComponent', () => {
    test('triggers onSearch with input value on Enter key', () => {
        const mockSearch = vi.fn();
        render(<SearchComponent onSearch={mockSearch} />);
        const input = screen.getByLabelText('Search');
        fireEvent.change(input, { target: { value: 'test' } });
        fireEvent.keyDown(input, { key: 'Enter' });
        expect(mockSearch).toHaveBeenCalledWith('test');
    });

    test('updates search text on input change', () => {
        render(<SearchComponent onSearch={vi.fn()} />);
        const input = screen.getByLabelText('Search');
        fireEvent.change(input, { target: { value: 'new value' } });
        expect(input.value).toBe('new value');
    });


    test('matches search component snapshot', () => {
        const { asFragment } = render(<SearchComponent onSearch={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('RememberMeCheckbox', () => {
    test('toggles checked state when clicked', () => {
        let checked = false;
        const setChecked = (val) => checked = val;
        render(<RememberMeCheckbox rememberMe={checked} setRememberMe={setChecked} />);
        const checkbox = screen.getByRole('checkbox');
        fireEvent.click(checkbox);
        expect(checked).toBe(true);
    });


    test('matches checkbox snapshot', () => {
        const { asFragment } = render(
            <RememberMeCheckbox
                rememberMe={false}
                setRememberMe={vi.fn()}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('getVisibleRows', () => {
    test('sorts and paginates correctly', () => {
        const rows = [{ id: 1, name: 'Z' }, { id: 2, name: 'A' }];
        const result = getVisibleRows(rows, 'asc', 'name', 0, 1);
        expect(result[0].name).toBe('A');
        expect(result.length).toBe(1);
    });


    test('matches sorted rows snapshot', () => {
        const rows = [{ id: 1, name: 'Z' }, { id: 2, name: 'A' }];
        const result = getVisibleRows(rows, 'asc', 'name', 0, 1);
        expect(result).toMatchSnapshot();
    });
});

describe('handleClick', () => {
    test('toggles selection correctly', () => {
        let selected = [];
        const setSelected = (newVal) => selected = newVal;

        handleClick(null, 1, selected, setSelected);
        expect(selected).toEqual([1]);

        handleClick(null, 1, selected, setSelected);
        expect(selected).toEqual([]);

        selected = [1, 2];
        handleClick(null, 2, selected, setSelected);
        expect(selected).toEqual([1]);

        selected = [1, 2, 3];
        handleClick(null, 2, selected, setSelected);
        expect(selected).toEqual([1, 3]);
    });


    test('matches selection states snapshot', () => {
        let selected = [];
        const setSelected = (newVal) => selected = newVal;

        // Test multiple states
        handleClick(null, 1, selected, setSelected);
        const state1 = [...selected];

        handleClick(null, 1, selected, setSelected);
        const state2 = [...selected];

        selected = [1, 2, 3];
        handleClick(null, 2, selected, setSelected);
        const state3 = [...selected];

        expect([state1, state2, state3]).toMatchSnapshot();
    });
});

describe('EnhancedTableHead', () => {
    const mockSetOrder = vi.fn();
    const mockSetOrderBy = vi.fn();
    const mockSetSelected = vi.fn();
    const defaultProps = {
        order: 'asc',
        orderBy: 'name',
        setOrder: mockSetOrder,
        setOrderBy: mockSetOrderBy,
        numSelected: 0,
        headCells: mockHeadCells,
        rows: mockRows,
        setSelected: mockSetSelected,
        idAttributeName: 'id'
    };

    afterEach(() => {
        vi.clearAllMocks();
    });

    test('renders select all checkbox correctly', () => {
        render(
            <table>
                <EnhancedTableHead
                    {...defaultProps}
                />
            </table>
        );
        const checkbox = screen.getByRole('checkbox');
        expect(checkbox).toBeInTheDocument();
    });

    test('selects all rows when checkbox is checked', () => {
        const { container } = render(<table><EnhancedTableHead {...defaultProps} /></table>);
        const checkbox = container.querySelector('input[type="checkbox"]');
        fireEvent.click(checkbox);
        expect(mockSetSelected).toHaveBeenCalledWith([1, 2]);
    });

    test('maintains sorting order when clicking different header', () => {
        const { getByText } = render(<table><EnhancedTableHead {...defaultProps} /></table>);
        const sortLabel = getByText('Name');
        fireEvent.click(sortLabel);
    });


    test('matches table head snapshot', () => {
        const { asFragment } = render(
            <table>
                <EnhancedTableHead
                    {...defaultProps}
                    numSelected={2}
                />
            </table>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('EnhancedTableToolbar', () => {
    test('shows delete button when items are selected', () => {
        render(
            <EnhancedTableToolbar
                numSelected={2}
                deleteSelected={vi.fn()}
                tableTitle="Test"
                onSearch={vi.fn()}
            />
        );
        expect(screen.getByLabelText(/delete/i)).toBeInTheDocument();
    });

    test('shows update button when one item is selected', () => {
        render(
            <EnhancedTableToolbar
                numSelected={1}
                deleteSelected={vi.fn()}
                tableTitle="Test"
                onSearch={vi.fn()}
            />
        );
        expect(screen.getByLabelText(/update/i)).toBeInTheDocument();
    });

    test('shows search field when no items are selected', () => {
        render(
            <EnhancedTableToolbar
                numSelected={0}
                tableTitle="Test"
                onSearch={vi.fn()}
            />
        );
        expect(screen.getByLabelText('Search')).toBeInTheDocument();
    });


    test('matches toolbar states snapshot', () => {
        const { asFragment: deleteState } = render(
            <EnhancedTableToolbar
                numSelected={2}
                deleteSelected={vi.fn()}
                tableTitle="Test"
            />
        );

        const { asFragment: updateState } = render(
            <EnhancedTableToolbar
                numSelected={1}
                deleteSelected={vi.fn()}
                tableTitle="Test"
            />
        );

        const { asFragment: searchState } = render(
            <EnhancedTableToolbar
                numSelected={0}
                tableTitle="Test"
            />
        );

        expect({ deleteState: deleteState(), updateState: updateState(), searchState: searchState() }).toMatchSnapshot();
    });
});

describe('PieChartCard', () => {
    const TEST_COLORS_LENGTH = pieChartColors.length;
    // Test case 1: Data exceeds color threshold
    it('should merge excess items into "Other" category when data exceeds color limit', () => {
        // Create data with 1 more item than available colors
        const longData = Array.from({ length: TEST_COLORS_LENGTH + 2 }, (_, i) => ({
            id: `item-${i}`,
            label: `Category ${i + 1}`,
            value: 1000 - i * 100, // Descending values
        }));

        render(<PieChartCard data={longData} title="Test Chart" showLinear={true} />);

        // Verify merged data
        const otherItem = screen.getByText('Other');
        expect(otherItem).toBeInTheDocument();

        // Verify original items count after merge (N-1 colors + Other)
        const slicedItems = longData.slice(0, TEST_COLORS_LENGTH - 1);
        slicedItems.forEach(item => {
            expect(screen.getByText(item.label)).toBeInTheDocument();
        });
    });

    // Test case 2: Data within color limit
    it('should display all items directly when data is within color limit', () => {
        // Create data with 2 items less than color limit
        const shortData = Array.from({ length: TEST_COLORS_LENGTH - 2 }, (_, i) => ({
            id: `item-${i}`,
            label: `Category ${i + 1}`,
            value: 1000 - i * 100,
        }));

        render(<PieChartCard data={shortData} title="Test Chart" showLinear={true} />);

        // Verify no "Other" category
        expect(screen.queryByText('Other')).not.toBeInTheDocument();

        // Verify all items are displayed
        shortData.forEach(item => {
            expect(screen.getByText(item.label)).toBeInTheDocument();
        });
    });

    // Optional: Add tests for total value calculation, sorting, etc.
    it('should calculate and display correct total value', () => {
        const testData = [
            { id: '1', label: 'A', value: 500 },
            { id: '2', label: 'B', value: 300 },
        ];

        render(<PieChartCard data={testData} title="Test Chart" showLinear={false} />);

        const totalValue = testData.reduce((sum, item) => sum + item.value, 0);
        expect(screen.getByText(totalValue.toFixed(2))).toBeInTheDocument();
    });
});