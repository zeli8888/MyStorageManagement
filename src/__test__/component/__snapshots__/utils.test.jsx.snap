// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeletionConfirmationComponent > matches dialog open snapshot 1`] = `<DocumentFragment />`;

exports[`EnhancedTableHead > matches table head snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-47m1dp-MuiTableCell-root"
          scope="col"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort="ascending"
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
          scope="col"
        >
          <span
            class="MuiButtonBase-root MuiTableSortLabel-root Mui-active MuiTableSortLabel-directionAsc css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
            role="button"
            tabindex="0"
          >
            Name
            <span
              class="MuiBox-root css-ty8jgw"
            >
              sorted ascending
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
              data-testid="ArrowDownwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;

exports[`EnhancedTableToolbar > matches toolbar states snapshot 1`] = `
{
  "deleteState": <DocumentFragment>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-gfg877-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-subtitle1 css-1hm74xa-MuiTypography-root"
      >
        2 selected
      </div>
      <button
        aria-label="Delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </div>
  </DocumentFragment>,
  "searchState": <DocumentFragment>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-l0qoig-MuiToolbar-root"
    >
      <div
        class="MuiStack-root css-94iksa-MuiStack-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 css-ch9kh4-MuiTypography-root"
          id="tableTitle"
        >
          Test
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1ucyl46-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rt:"
            id=":rt:-label"
          >
            Search
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1oe8v0m-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rt:"
              type="text"
              value=""
            />
            <button
              aria-label="search"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </button>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1n64csd-MuiNotchedOutlined-root"
              >
                <span>
                  Search
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </DocumentFragment>,
  "updateState": <DocumentFragment>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-gfg877-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-subtitle1 css-1hm74xa-MuiTypography-root"
      >
        1 selected
      </div>
      <button
        aria-label="Update"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.996.996 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
          />
        </svg>
      </button>
      <button
        aria-label="Delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </div>
  </DocumentFragment>,
}
`;

exports[`RememberMeCheckbox > matches checkbox snapshot 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-14r958l-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-reoaak-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
        data-indeterminate="false"
        name="remember"
        type="checkbox"
        value="true"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1c26xtj-MuiTypography-root"
    >
      Remember me
    </span>
  </label>
</DocumentFragment>
`;

exports[`SearchComponent > matches search component snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-14dcxqn-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":rd:"
      id=":rd:-label"
    >
      Search
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1oe8v0m-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
        id=":rd:"
        type="text"
        value=""
      />
      <button
        aria-label="search"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
      </button>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1n64csd-MuiNotchedOutlined-root"
        >
          <span>
            Search
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`getVisibleRows > matches sorted rows snapshot 1`] = `
[
  {
    "id": 2,
    "name": "A",
  },
]
`;

exports[`handleClick > matches selection states snapshot 1`] = `
[
  [
    1,
  ],
  [],
  [
    1,
    3,
  ],
]
`;
